File: app/src/main/java/com/yfbx/widgets/fragment/DrawFragment.java
package com.yfbx.widgets.fragment;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.ImageView;

import com.yfbx.widgets.R;
import com.yfbx.widgets.util.BitmapUtil;
import com.yfbx.widgets.util.PxUtil;

import butterknife.BindView;

/**
 * Author: Edward
 * Date: 2019/1/7
 * Description:
 */


public class DrawFragment extends BaseFragment {

    @BindView(R.id.img)
    ImageView img;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        draw();
    }

    @Override
    public int getLayout() {
        return R.layout.frag_draw;
    }


    private void draw() {
        //文件容器
        Bitmap bitmap = Bitmap.createBitmap(256, 256, Bitmap.Config.ARGB_8888);
        bitmap.eraseColor(0x00FFFFFF);//背景透明，默认黑色

        //画布
        Canvas canvas = new Canvas(bitmap);
        //画笔
        Paint paint = new Paint();
        paint.setStyle(Paint.Style.FILL);
        paint.setAntiAlias(true);

        //绘制
        canvas.translate(128, 128);
        paint.setColor(Color.WHITE);
        canvas.drawCircle(0, 0, 100, paint);

        paint.setTextSize(PxUtil.sp(36));
        paint.setColor(Color.BLUE);
        Rect rect = new Rect();
        String text = "E";
        paint.getTextBounds(text, 0, text.length(), rect);
        canvas.drawText(text, -rect.width() / 2, rect.height() / 2, paint);

        //展示
        img.setImageBitmap(bitmap);

        //保存
        BitmapUtil.saveBitmap(bitmap, "test");

    }


}

File: app/src/main/java/com/yfbx/widgets/fragment/DrawFragment.kt
package com.yfbx.widgets.fragment

import android.graphics.*
import android.os.Bundle
import android.view.View
import com.yfbx.widgets.R
import com.yfbx.widgets.util.saveCapture
import com.yfbx.widgets.util.sp
import kotlinx.android.synthetic.main.frag_draw.*


/**
 * Author: Edward
 * Date: 2019/1/7
 * Description:绘制图片
 */
class DrawFragment : BaseFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        draw()

        saveBtn.setOnClickListener {
            img.saveCapture()
        }
    }

    override fun getLayout(): Int {
        return R.layout.frag_draw
    }

    private fun draw() {
        //文件容器
        val bitmap = Bitmap.createBitmap(256, 256, Bitmap.Config.ARGB_8888)
        bitmap.eraseColor(0x00FFFFFF)//背景透明，默认黑色

        //画布
        val canvas = Canvas(bitmap)
        //画笔
        val paint = Paint()
        paint.style = Paint.Style.FILL
        paint.isAntiAlias = true

        //绘制
        canvas.translate(128f, 128f)
        paint.color = Color.WHITE
        canvas.drawCircle(0f, 0f, 100f, paint)

        paint.textSize = sp(36).toFloat()
        paint.color = Color.BLUE
        val rect = Rect()
        val text = "E"
        paint.getTextBounds(text, 0, text.length, rect)
        canvas.drawText(text, (-rect.width() / 2).toFloat(), (rect.height() / 2).toFloat(), paint)

        //展示
        img.setImageBitmap(bitmap)

    }


}
